### PR TITLE
Fix #1584 color graphing to loop through blue, red, green.

### LINKED ISSUE #1584: Issues with color plotting for deep workflows
For deeply nested workflows (5 deep in my case), colors don't seem to operate as intended.

Sometimes (but perplexingly, not always) an error is thrown when `Workflow._get_dot` indexes beyond the length of `colorset`.

### PR DESCRIPTION
**Pull request by ashgillman**
suggested to Fix #1584

Colors now cycle, yellow for top-level, then B, R, G, B, R, G...

### CODE REVIEW

**Comment by ashgillman** [comment]
I realised this PR was very unclear, so I have added failing tests demonstrating the issue which now pass.

**Comment by ashgillman** [comment]
Whoops, looks like CI test don't have dot. Is there a way I can test producing the dot file without running graphviz to produce a png?
Actually, I thought dot was only run if it was present on the system?

**Comment by oesteban** [comment]
@ashgillman - apparently, the problem only affect travis. It makes sense since I think I included graphviz in the docker image (which is what circleci uses).

You'll need to install graphviz, in the .travis.yml file.

File: .travis.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ install:
   conda config --add channels conda-forge &&
   conda update --yes conda &&
   conda update --all -y python=$TRAVIS_PYTHON_VERSION &&
-  conda install -y nipype &&
+  conda install -y nipype graphviz &&
```
Comment (oesteban) [inline .travis.yml]: I think that should go here: https://github.com/ashgillman/nipype/blob/4c3a232b5d70da3638ff54dea173bcf3adf55acb/.travis.yml#L22 as an apt package.

Comment (ashgillman) [inline .travis.yml]: Ah, yes that would make more sense wouldn't it

**Comment by ashgillman** [comment]
Is the remaining error due to my code or a run time thing? I can't really tell

**Comment by chrisfilo** [comment]
I have restarted the build hopefully the error will go away. We are working on a fix.

**Comment by ashgillman** [comment]
Looks to have times out for Py27 (but not Py35) at test

    nipype.pipeline.plugins.tests.test_callback.test_callback_multiproc_exception('f_node', 'f_node') ... ok

Not sure if this test is problematic or whether that test just happened to be running during the timeout

**Comment by ashgillman** [comment]
Also I don't know if the Travis has rerun since I updated travis.yml

**Pull request merged**